xclim/indices/_agro.py

```diff
@@ -1418,3 +1419,75 @@ def effective_growing_degree_days(
     egdd = aggregate_between_dates(deg_days, start=start, end=end, freq=freq)
 
     return to_agg_units(egdd, tas, op="delta_prod")
+
+
+@declare_units(tasmin="[temperature]")
+def hardiness_zones(
+    tasmin: xarray.DataArray, window: int = 30, method: str = "usda", freq: str = "YS"
+):
+    """Hardiness zones.
+
+    Hardiness zones are a categorization of the annual extreme temperature minima, averaged over a certain period.
+    The USDA defines 14 zones, each divided into two sub-zones, using steps of 5°F, starting at -65°F.
+    The Australian National Botanic Gardens define 7 zones, using steps of 5°C, starting at -15°C.
+
+    Parameters
+    ----------
+    tasmin : xr.DataArray
+        Minimum temperature.
+    window : int
+        The length of the averaging window, in years.
+    method : {'usda', 'anbg'}
+        Whether to return the American (`usda`) or the Australian (`anbg`) classification zones.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+        {method} hardiness zones.
+        US sub-zones are denoted by using a half step. For example, Zone 4b is given as 4.5.
+        Values are given at the end of the averaging window.
+
+    References
+    ----------
+    :cite:cts:`usda_2012`
+    :cite:cts:`dawson_plant_1991`
+    """
+    # Construct climatology
+    if parse_offset(freq) != (1, "A", True, "JAN"):
+        raise ValueError(f"Freq not allowed: {freq}. Must be `YS` or `AS-JAN`")
+
+    tnmin = tasmin.resample(time=freq).min().rolling(time=window).mean()
+
+    if method.lower() == "usda":
+        tnmin = convert_units_to(tnmin, "degF")
+        bins = np.append(
+            np.insert(np.arange(-65, 66, 5).astype("float"), 0, -np.inf), np.inf
+        )
```

Comment:
I suspect that for very large datasets, it would be faster to convert the units of the bins to the tasmin units instead.

Reply:
That's a good point. We should probably be careful with the size of bins. To conserve the size of bins: 
```
        b0, b1 = "-65 degF", "65 degF"
        bins = np.linspace(convert_units_to(b0,tmin), convert_units_to(b1,tmin), int(1+(2*65)/5))
        bins = np.append(np.insert(bins, 0, -np.inf), np.inf)
```
and similarly for the degC part

Reply:
A more similar implementation with more call to `convert_units_to`  could be: 
```
bins = np.arange(-65, 66, 5)
bins = np.array([convert_units_to(str(bi) + " degF", tmin) for bi in bins])
bins = np.append(np.insert(bins, 0, -np.inf), np.inf)
```